File: backend/app/models/booking.py
from sqlalchemy import ForeignKey, String
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.domain.enums import BookingStatus
from app.models.base import Base, CreatedAtMixin
from app.models.lot import Lot


class Booking(Base, CreatedAtMixin):
    __tablename__ = "bookings"

    id: Mapped[int] = mapped_column(primary_key=True)
    lot_id: Mapped[int] = mapped_column(ForeignKey("lots.id", ondelete="RESTRICT"), index=True)
    name: Mapped[str] = mapped_column()
    contact: Mapped[str] = mapped_column()  # телефон или почта
    status: Mapped[BookingStatus] = mapped_column(String(16), default=BookingStatus.ACTIVE)

    lot: Mapped[Lot] = relationship(lazy="selectin")

File: frontend/src/domain/booking.ts
import type { Lot } from './lot'

export const BOOKING_STATUSES = ['active', 'cancelled'] as const
export type BookingStatus = (typeof BOOKING_STATUSES)[number]

export const BOOKING_STATUS_LABELS: Record<BookingStatus, string> = {
  active: 'Активна',
  cancelled: 'Отменена',
}

export interface Booking {
  id: number
  lotId: number
  name: string
  contact: string
  status: BookingStatus
  createdAt: Date
  lot: Lot | null
}

export interface BookingInput {
  lotId: number
  name: string
  contact: string
}

export interface BookingQuery {
  status?: BookingStatus
  page: number
  pageSize: number
}
